src/swarm/neo/util/AcquiredResources.d

```diff
@@ -372,6 +372,167 @@ unittest
 
 /*******************************************************************************
 
+    Singleton (per-request) acquired resources of the templated type. An
+    external source of elements of this type -- a FreeList!(T) -- is required.
+    When the singleton resource is acquired (via the acquire() method), it is
+    requested from the free list and stored internally. All subsequent calls to
+    acquire() return the same instance. When the resource is nolonger required,
+    the relinquish() method will return it to the free list.
+
+    Params:
+        T = type of resource
+
+*******************************************************************************/
+
+public struct AcquiredSingleton ( T )
+{
+    import ocean.util.container.pool.FreeList;
+
+    /***************************************************************************
+
+        Determine the type of a new resource.
+
+    ***************************************************************************/
+
+    static if ( is(typeof({T* t = new T;})) )
```

Comment:
Yeah, I guess that would have the same result. I just copied what's being used already in this module (https://github.com/gavin-norman-sociomantic/swarm-tsunami/blob/9b63f1077d8ae0c2cfa64c270ed99124185f0a2d/src/swarm/neo/util/AcquiredResources.d#L201).